refactor: rewrite Flox in 5 Minutes (#316)
flox/floxdocs: docs/flox-5-minutes.md

# Flox in 5 minutes

Flox is a next-generation, language-agnostic package and environment manager.

> With it you create sets of tools, environment variables, and setup scripts that work reproducibly from machine to machine, x86_64 to Arm, and macOS to Linux.

Reviewer: Let's keep "stack" in this paragraph so that the next paragraph flows better.

```suggestion
With it you create stacks of tools, environment variables, and setup scripts that work reproducibly from machine to machine, x86_64 to Arm, and macOS to Linux.
```

> The cherry on top is that all of this works without causing version conflicts between projects.

Reviewer: Idioms can trip up non-english speakers, not sure how universal this one is.

```suggestion
The best part is that all of this works without causing version conflicts between projects.
```

> We call these stacks **"Flox environments"**.

Reviewer: I don't think we need both quotes and bold here. My personal preference is to use bold to introduce new terms (less symbol clutter than quotes).

```suggestion
We call these stacks **Flox environments**.
```

Author: Fixed

Flox environments are based on carefully configured subshells, so there's no container isolation preventing you from using your favorite tools or artisanally handcrafted dotfiles.
Even better, these environments _compose_ and _layer_ so you can prepare different environments
for different needs and combine them to seamlessly work across different contexts.

Finally, you can use Flox environments for local development, CI, and production to ensure that you have a consistent set of software across the entire software development lifecycle.

Buckle up, it's time for a whirlwind tour of Flox.

## Get the project

We've prepared a sample project for you, but you'll need to [install Flox][install_flox] to follow along.
Once you have Flox, you can clone the project:

> <!-- markdownlint-disable MD014 -->
> ```console
> $ git clone https://github.com/flox/flox-in-5min.git
> $ cd flox-in-5min
> ```

Reviewer: Given the change in https://github.com/flox/floxdocs/pull/311, let's format to be copy/paste friendly:

````suggestion
```{ .bash .copy }
git clone https://github.com/flox/flox-in-5min.git
cd flox-in-5min
```
````

Author: This won't copy properly without a semicolon at the end of the line, and I kind of hate how that looks. I think it's totally fine to have copyable one-liners, but it doesn't work well for multiline code blocks.

Reviewer: Fair enough, let's table this for now.

Author: I'm wrong. I fixed this in a couple of places.

## Tools prepared for you

Once you have the project, you can run the [`flox activate`][activate] command to enter the environment:

> ```console
> $ flox activate
> ✅ You are now using the environment 'flox-in-5min'.
> To stop using this environment, type 'exit'
> ```

Reviewer: @brendaneamon and I discussed splitting inputs/outputs into different code blocks so that you can more easily copy the inputs into your terminal.

````suggestion
```{ .bash .copy }
flox activate
```
```console
✅ You are now using the environment 'flox-in-5min'.
To stop using this environment, type 'exit'
```
````

emacsPackages.ripgrep             <no description provided>
vimPlugins.blink-ripgrep-nvim     <no description provided>
emacsPackages.projectile-ripgrep  <no description provided>
vgrep                             User-friendly pager for grep/git-grep/ripgrep
repgrep                           Interactive replacer for ripgrep that makes it easy to find and replace across files on the command line
grip-grab                         Fast, more lightweight ripgrep alternative for daily use cases
bat-extras.batgrep                Quickly search through and highlight files using ripgrep

Use 'flox show <package>' to see available versions
```

Cool, what if I want a specific version?

```console
$ flox show ripgrep
ripgrep - Utility that combines the usability of The Silver Searcher with the raw speed of grep
    ripgrep@14.1.1
    ripgrep@14.1.0
    ripgrep@14.0.3
    ripgrep@14.0.1
    ripgrep@13.0.0
    ripgrep@12.1.1 (aarch64-linux, x86_64-darwin, x86_64-linux only)
```

## What else can Flox do?

Your environment is stored as a `.flox` directory in your repository, but you can also [`push`][push] it to [FloxHub][floxhub] to make it centrally managed and available from anywhere.

> We only have 5 minutes, so we're going to skip over that for now, but see the [sharing environments tutorial][sharing] for more information.

Reviewer: Love the FloxHub teaser here.
